apps/research-service/src/polygon-clipping/polygon-clipping.controller.ts

```diff
@@ -5,108 +5,189 @@ import {
   Post,
   UnauthorizedException,
   BadRequestException,
-  Param
+  Param,
+  Request
 } from "@nestjs/common";
 import { ApiOperation, ApiTags } from "@nestjs/swagger";
-import { ExceptionResponse } from "@terramatch-microservices/common/decorators";
+import { ExceptionResponse, JsonApiResponse } from "@terramatch-microservices/common/decorators";
 import { PolygonClippingService } from "./polygon-clipping.service";
 import { PolygonListClippingRequestBody } from "./dto/clip-polygon-request.dto";
 import { PolicyService } from "@terramatch-microservices/common";
-import { SitePolygon } from "@terramatch-microservices/database/entities";
+import { DelayedJob, Site, SitePolygon, User } from "@terramatch-microservices/database/entities";
+import { InjectQueue } from "@nestjs/bullmq";
+import { Queue } from "bullmq";
+import { DelayedJobDto } from "@terramatch-microservices/common/dto/delayed-job.dto";
+import { buildJsonApi } from "@terramatch-microservices/common/util";
+import { ClippedVersionDto } from "./dto/clipped-version.dto";
+import { populateDto } from "@terramatch-microservices/common/dto/json-api-attributes";
 
 @ApiTags("Polygon Clipping")
 @Controller("polygonClipping/v3")
 export class PolygonClippingController {
   constructor(
     private readonly clippingService: PolygonClippingService,
-    private readonly policyService: PolicyService
+    private readonly policyService: PolicyService,
+    @InjectQueue("clipping") private readonly clippingQueue: Queue
   ) {}
 
-  // These endpoints do NOT modify the database as we do not create new versions yet. It is here for testing clipping logic.
-
-  @Post("sites/:siteUuid/clippedPolygons")
+  @Post("sites/:siteUuid/clippedVersions")
   @ApiOperation({
-    operationId: "createSitePolygonClipping",
-    summary: "Create polygon clipping for a site",
+    operationId: "createSiteClippedVersions",
+    summary: "Create new polygon versions with clipped geometries for a site",
     description: `Finds and clips all fixable overlapping polygons in a site (overlap ≤3.5% AND ≤0.118 hectares).
-      Returns GeoJSON of original and clipped polygons for verification.`
+      Creates new versions asynchronously with clipped geometries. Returns a delayed job to track progress.`
   })
+  @JsonApiResponse(DelayedJobDto)
```

Comment:
This should include the DTO that will eventually be returned so the type ends up in the client. See https://github.com/wri/terramatch-microservices/blob/staging/apps/research-service/src/validations/validation.controller.ts#L135